sodym/trade.py

```diff
@@ -0,0 +1,79 @@
+import sys
+from pydantic import BaseModel as PydanticBaseModel
+from pydantic import model_validator
+from abc import abstractmethod
+from .named_dim_arrays import Parameter
+from .dimensions import DimensionSet
+
+class Trade(PydanticBaseModel):
+    """ A TradeModule handles the storing and calculation of trade data for a given MFASystem."""
+
+    dims : DimensionSet
+    imports : Parameter
+    exports : Parameter
+
+    @model_validator(mode='after')
+    def validate_region_dimension(self):
+        assert 'r' in self.imports.dims.letters, "Imports must have a Region dimension."
+        assert 'r' in self.exports.dims.letters, "Exports must have a Region dimension."
+
+        return self
+
+    def __getitem__(self, key):
+        if key == 'Imports':
+            return self.imports
+        elif key == 'Exports':
+            return self.exports
+        else:
+            raise KeyError(f"Key {key} not found in Trade data - has to be either 'Imports' or 'Exports'.")
+
+    def __setitem__(self, key, value):
+        if key == 'Imports':
+            self.imports = value
+        elif key == 'Exports':
+            self.exports = value
+        else:
+            raise KeyError(f"Key {key} has to be either 'Imports' or 'Exports'.")
+
+class BalancedTrade(Trade):
+
+    @abstractmethod
+    def balance(self, **kwargs):
+        pass
```

Comment:
Potentially would be nice to have this with an `inplace: bool=True` option, along with the **kwargs. I guess it could be useful for debugging to make the Trade that has been balanced as a separate object from the starting unbalanced Trade object. But, this makes me think that we don't need it as an abstractmethod. Still undecided tbh. What do you think?

Reply:
Another option could be to call it '_balance' instead of 'balance_function' to make it clear that one is for public use and the other for internal use.

Reply:
Looks great :)